### PR TITLE
Use new formatting options in roslyn itself.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <RoslynDiagnosticsNugetPackageVersion>3.3.3-beta1.21105.3</RoslynDiagnosticsNugetPackageVersion>
     <MicrosoftCodeAnalysisNetAnalyzersVersion>6.0.0-preview1.21054.10</MicrosoftCodeAnalysisNetAnalyzersVersion>
     <MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.20623.3</MicrosoftCodeAnalysisTestingVersion>
-    <CodeStyleAnalyzerVersion>3.9.0</CodeStyleAnalyzerVersion>
+    <CodeStyleAnalyzerVersion>3.10.0-3.final</CodeStyleAnalyzerVersion>
```
[the remark this answers is not in the document]
Comment (jmarolf) [reply eng/Versions.props:33]: These packages do not exist on nuget.org. Is that work pending?

File: .editorconfig [hunk 2]
```diff
@@ -61,6 +61,14 @@ dotnet_style_coalesce_expression = true:suggestion
 dotnet_style_null_propagation = true:suggestion
 dotnet_style_explicit_tuple_names = true:suggestion
 
+# Whitespace options
+dotnet_style_allow_multiple_blank_lines_experimental = false:warning
+dotnet_diagnostic.IDE2000.severity = warning
+
+dotnet_style_allow_statement_immediately_after_block_experimental = false:warning
+dotnet_diagnostic.IDE2003.severity = warning
+
```
Comment (jcouv) [inline .editorconfig:70]: nit: extra blank line

Comment (CyrusNajmabadi) [reply .editorconfig:70]: :)

File: .editorconfig [hunk 3]
```diff
@@ -61,6 +61,14 @@ dotnet_style_coalesce_expression = true:suggestion
 dotnet_style_null_propagation = true:suggestion
 dotnet_style_explicit_tuple_names = true:suggestion
 
+# Whitespace options
+dotnet_style_allow_multiple_blank_lines_experimental = false:warning
+dotnet_diagnostic.IDE2000.severity = warning
+
+dotnet_style_allow_statement_immediately_after_block_experimental = true:none
+dotnet_diagnostic.IDE2003.severity = none
+
```
Comment (CyrusNajmabadi) [inline .editorconfig:70]:
```suggestion
```

File: .editorconfig [hunk 4]
```diff
@@ -182,6 +190,21 @@ csharp_indent_case_contents_when_block = true
...
+dotnet_diagnostic.IDE2002.severity = warning
+
+csharp_style_allow_blank_line_after_colon_in_constructor_initializer_experimental = false:warning
+dotnet_diagnostic.IDE2004.severity = warning
+
+
+# IDE0059: MultipleBlankLinesDiagnosticId
+dotnet_diagnostic.IDE0059.severity = warning
+
+
```
Comment (CyrusNajmabadi) [inline .editorconfig:207]:
```suggestion
```

File: src/Compilers/.editorconfig [hunk 5]
```diff
@@ -12,6 +12,15 @@ dotnet_diagnostic.RS0101.severity = none
 # RS0102: Braces must not have blank lines between them
 dotnet_diagnostic.RS0102.severity = none
 
+# IDE0059: Unnecessary assignment to a value
+dotnet_diagnostic.IDE0059.severity = none
```
Comment (CyrusNajmabadi) [inline src/Compilers/.editorconfig:16]: this was unfortunate.  but without this the compiler had tons of warnings here that touched a lot of core code.  i think we should take a followup PR that enables this in isolation.

File: eng/Versions.props (hunk 1 above)
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply eng/Versions.props:33]: @sharwell do you know the answer to this?  Who should we pull in to resolve this bit?

File: src/Compilers/.editorconfig (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/Compilers/.editorconfig:16]: @sharwell @jcouv for thoughts on this.  I really don't like the idea of disabling this (as i think it reveals many interesting cases where the code seems fishy).  But i don't want to have this PR go in witha  bunch of complex code changes that have to be closely reviewed and which carries the risk of a subtle bug being introduced.

File: src/Compilers/.editorconfig [hunk 6]
```diff
@@ -12,6 +12,15 @@ dotnet_diagnostic.RS0101.severity = none
...
+
+dotnet_diagnostic.IDE2000.severity = none
+dotnet_diagnostic.IDE2001.severity = none
+dotnet_diagnostic.IDE2002.severity = none
+dotnet_diagnostic.IDE2003.severity = none
+dotnet_diagnostic.IDE2004.severity = none
```
Comment (CyrusNajmabadi) [inline src/Compilers/.editorconfig:22]: compiler was disabling the RS.... diagnostics above.  It seems liek a pity, but i think we have to do the same with the IDE2XXX diagnostics until we can decide if we want to be consistent here across the codebase or not.

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpEESymbolProvider.cs [hunk 7]
```diff
@@ -102,8 +102,7 @@ public override IAssemblySymbolInternal GetReferencedAssembly(AssemblyReferenceH
         /// <exception cref="UnsupportedSignatureContent"></exception>
         public override TypeSymbol GetType(EntityHandle handle)
         {
-            bool isNoPiaLocalType;
-            return _metadataDecoder.GetSymbolForTypeHandleOrThrow(handle, out isNoPiaLocalType, allowTypeSpec: true, requireShortForm: false);
+            return _metadataDecoder.GetSymbolForTypeHandleOrThrow(handle, out _, allowTypeSpec: true, requireShortForm: false);
```
Comment (CyrusNajmabadi) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpEESymbolProvider.cs:105]: the EE is not under the compiler and doesn't ahve its own editorconfig.  So i did fix these all up.  THere were few enough cases that i felt ok doing this.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/AccessibilityTests.cs [hunk 8]
```diff
@@ -33,14 +33,13 @@ void M()
...
             string error;
-            var testData = Evaluate(
+            _ = Evaluate(
                 source,
                 OutputKind.DynamicallyLinkedLibrary,
                 methodName: "C.M",
                 expr: "this.get_P()",
-                resultProperties: out resultProperties,
+                resultProperties: out _,
                 error: out error);
```
Comment (CyrusNajmabadi) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/AccessibilityTests.cs:43]: most fixes were in tests.

File: .editorconfig [hunk 9]
```diff
@@ -61,6 +61,13 @@ dotnet_style_coalesce_expression = true:suggestion
 dotnet_style_null_propagation = true:suggestion
 dotnet_style_explicit_tuple_names = true:suggestion
 
+# Whitespace options
+dotnet_style_allow_multiple_blank_lines_experimental = false:warning
+dotnet_diagnostic.IDE2000.severity = warning
+
+dotnet_style_allow_statement_immediately_after_block_experimental = true:none
+dotnet_diagnostic.IDE2003.severity = none
```
Comment (CyrusNajmabadi) [inline .editorconfig:69]: would like to set this to 'false'.  but will come in a followup PR.

File: src/Compilers/.editorconfig (hunk 5 above)
Comment (sharwell) [reply src/Compilers/.editorconfig:16]: Is IDE0059 new for 3.10? I throught we disabled it for the compiler code a long time ago.

Comment (CyrusNajmabadi) [reply src/Compilers/.editorconfig:16]: Not sure. @mavasani ?  The diagnostics look appropriate. But there are a lot of them and I don't think we should do this without explicit compiler desire ahead of time.

File: eng/Versions.props (hunk 1 above)
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply eng/Versions.props]: this has been completed.

File: src/Compilers/.editorconfig (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/Compilers/.editorconfig:16]: @mavasani any ideas here?

**Comment by CyrusNajmabadi** [comment]
@sharwell does thsi need anything else (other than reviews of course) to be able to move forward?

File: .editorconfig (hunk 9 above)
Comment (sharwell) [reply .editorconfig:69]: Can we set it to `false:suggestion` for this PR, and just not fix the current reports yet?

File: .editorconfig [hunk 10]
```diff
@@ -182,6 +189,16 @@ csharp_indent_case_contents_when_block = true
...
 
+# Whitespace options
+csharp_style_allow_embedded_statements_on_same_line_experimental = false:warning
+dotnet_diagnostic.IDE2001.severity = warning
+
+csharp_style_allow_blank_lines_between_consecutive_braces_experimental = false:warning
+dotnet_diagnostic.IDE2002.severity = warning
+
+csharp_style_allow_blank_line_after_colon_in_constructor_initializer_experimental = false:warning
+dotnet_diagnostic.IDE2004.severity = warning
```
Comment (sharwell) [inline .editorconfig:200]: ❗ All of the warning diagnostics need to move under the IDE section (starting line 257 in this file) unless the compiler separately signed off on enabling them.

**Review by sharwell — CHANGES_REQUESTED** [review]

File: src/Compilers/.editorconfig (hunk 5 above)
Comment (sharwell) [reply src/Compilers/.editorconfig:16]: > but without this the compiler had tons of warnings

This cannot be correct. Our editorconfig does not set IDE0059 to warning for the compiler folder.

File: src/Compilers/.editorconfig (hunk 6 above)
Comment (sharwell) [inline src/Compilers/.editorconfig:22]: All of these lines can be removed.

* IDE0059 is not set to warning for the compiler folder at the higher level, so the override here is unnecessary
* IDE2000-IDE2004 should only be set for the IDE folders at the higher level, so the overrides here are unnecessary

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Binders/PlaceholderLocalBinder.cs [hunk 11]
```diff
@@ -30,8 +30,8 @@ internal sealed class PlaceholderLocalBinder : LocalScopeBinder
             ImmutableArray<Alias> aliases,
             MethodSymbol containingMethod,
             EETypeNameDecoder typeNameDecoder,
-            Binder next) :
-            base(next)
+            Binder next)
+            : base(next)
```
Comment (sharwell) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Binders/PlaceholderLocalBinder.cs:34]: src/ExpressionEvaluator should not have been included in this PR.

File: src/Dependencies/PooledObjects/PooledHashSet.cs [hunk 12]
```diff
@@ -13,8 +13,8 @@ internal sealed partial class PooledHashSet<T> : HashSet<T>
     {
         private readonly ObjectPool<PooledHashSet<T>> _pool;
 
-        private PooledHashSet(ObjectPool<PooledHashSet<T>> pool, IEqualityComparer<T> equalityComparer) :
-            base(equalityComparer)
+        private PooledHashSet(ObjectPool<PooledHashSet<T>> pool, IEqualityComparer<T> equalityComparer)
+            : base(equalityComparer)
```
Comment (sharwell) [inline src/Dependencies/PooledObjects/PooledHashSet.cs:17]: src/Dependencies should not have been included in this PR

File: src/Interactive/HostTest/InteractiveHostDesktopTests.cs [hunk 13]
```diff
@@ -435,7 +435,7 @@ public async Task AddReference_Dependencies_DllExe()
             var dir = Temp.CreateDirectory();
 
             var dll = CompileLibrary(dir, "c.dll", "C", @"public class C { public static int Main() { return 1; } }");
-            var exe = CompileLibrary(dir, "c.exe", "C", @"public class C { public static int Main() { return 2; } }");
```
Comment (sharwell) [inline src/Interactive/HostTest/InteractiveHostDesktopTests.cs:438]: src/Interactive should not have been included in the scope for new warnings

**Review by sharwell — APPROVED** [review]

**Pull request merged**